backend/kernelCI_app/management/commands/process_pending_aggregations.py

```diff
@@ -0,0 +1,425 @@
+        "boot_failed": 0,
+        "boot_inc": 0,
+        "test_pass": 0,
+        "test_failed": 0,
+        "test_inc": 0,
+    }
+
+
+def _collect_test_contexts(
+    tests_instances: Sequence[PendingTest],
+    builds_by_id: dict[str, Builds],
+) -> tuple[list[tuple[PendingTest, Builds, Checkouts, bytes]], set[bytes]]:
+    """Collect valid test contexts with their associated build and checkout."""
+    contexts = []
+    keys_to_check = set()
+
+    for test in tests_instances:
+        try:
+            build = builds_by_id[test.build_id]
+            checkout = build.checkout
+            if not checkout:
+                continue
+        except KeyError:
+            continue
+
+        h_key = get_hardware_key(test.origin, test.platform, checkout.id)
+        contexts.append((test, build, checkout, h_key))
+        keys_to_check.add(h_key)
+
+    return contexts, keys_to_check
+
+
+def _get_existing_processed(keys_to_check: set[bytes]) -> set[ProcessedHardwareStatus]:
+    """Fetch existing processed entries from the database."""
+    return set(ProcessedHardwareStatus.objects.filter(hardware_key__in=keys_to_check))
+
+
+def _process_test_status(
+    test: PendingTest,
+    h_key: bytes,
+    status_record: HardwareStatusRecord,
+    existing_processed: set[ProcessedHardwareStatus],
+    new_processed_entries: set[ProcessedHardwareStatus],
+) -> bool:
+    """Process test status and update status record if not already processed."""
+    to_process = ProcessedHardwareStatus(
+        hardware_key=h_key,
+        entity_id=test.test_id,
+        entity_type=HardwareStatusEntityType.TEST,
+    )
+
+    if to_process in existing_processed:
+        return False
+
+    t_pass, t_fail, t_inc = map_simplified_status_to_count(test.status)
+
+    if test.is_boot:
+        status_record["boot_pass"] += t_pass
+        status_record["boot_failed"] += t_fail
+        status_record["boot_inc"] += t_inc
+    else:
+        status_record["test_pass"] += t_pass
+        status_record["test_failed"] += t_fail
+        status_record["test_inc"] += t_inc
+
+    new_processed_entries.add(to_process)
+
+    return True
+
+
+def _process_build_status(
+    build: Builds,
+    h_key: bytes,
+    status_record: HardwareStatusRecord,
+    existing_processed: set[ProcessedHardwareStatus],
+    new_processed_entries: set[ProcessedHardwareStatus],
+) -> None:
+    """Process build status and update status record if not already processed."""
+
+    """
+    We can't filter out dummy builds before this step,
+    because although we don't count them in build_* status,
+    there are tests that are associated with them, and those tests are counted in test_* status.
+    """
+    if build.id.startswith(MAESTRO_DUMMY_BUILD_PREFIX):
+        return
+
+    to_process = ProcessedHardwareStatus(
+        hardware_key=h_key,
+        entity_id=build.id,
+        entity_type=HardwareStatusEntityType.BUILD,
+    )
+
+    if to_process in existing_processed:
+        return
```

Comment:
when we move to `hardware_key` to include entity_id/type, then `build_h_key` is already what we'll check in `existing_processed`, no need to create the model instance here.